Services/Certificate/classes/File/Template/class.ilCertificatePathFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/* Copyright (c) 1998-2019 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * @author  Niels Theen <ntheen@databay.de>
+ */
+class ilCertificatePathFactory
+{
+	/**
+	 * @param ilObject $object
+	 * @return string
+	 * @throws ilException
+	 */
+	public function createCertificatePath(ilObject $object)
```

Comment:
For the future I would prefer a single place where a consuming object could register itself and is able to provide paths, adapters etc.